### ISSUE TITLE
state_color doesn't work with switch template within groups

### ISSUE THREAD

**Issue by xenz698** [home-assistant/frontend#5784]
## Checklist

- [X ] I have updated to the latest available Home Assistant version.
- [X ] I have cleared the cache of my browser.
- [X ] I have tried a different browser to see if it is related to my browser.

## The problem

When I put a switch template entity_id into lovelace entities card, the state_color of the icon work OK (ON=yellow, OFF= grey). when I put it inside a group, and put this group entity_id within lovelace entities card. the switch template state_color doesn't work (ON=grey, OFF=grey).

## Expected behavior

when putting this switch template entity_ID within group. and put this group entity_ID as lovelace entities card, the state_color should turn yellow when ON, and grey when OFF

## Steps to reproduce

1. create a simple switch template with value_template (switch.garagedoor)
2. add entity_ID to lovelace card as an entity card,  it should just show an on/off switch
3. add state_color to this card, ON will be yellow, OFF will be grey
4. add this switch template entity_ID to group.yaml (test_group)
5. add this group entity_ID (group.test_group) to lovelace card as an entity card, it should just show an on/off switch
6. add state_color to this card
7. click on group name, a popup window will show the "more-info" page with the listed entity at the bottom.
8. this entity state color will not change.

## Environment

- Home Assistant release with the issue: tried 0.108.2, 0.108.7, 0.109.4
- Last working Home Assistant release (if known): before 105 I believe, back then don't need state_color
- Browser and browser version: Chrome (81.0.4044.129), Firefox (76.0)
- Operating system: Windows 7 SP1, Windows 10 1809

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```
switch.yaml
- platform:template
  switches:
    garagedoor:
      friendly_name: "Garage Door"
      value_template: '{{ is_state("cover.garagedoor","open") }}'
      turn_on:
        - service: cover.open_cover
          data: 
            entity_id: cover.garagedoor
      turn_off:
        - service: cover.close_cover
          data: 
            entity_id: cover.garagedoor


groups.yaml
test_group:
  name: test group
  entities:
  - switch.garagedoor


ui-lovelace.yaml
views:
  - title: Test
    cards:
    - type: entities
      title: Test
      show_header_toggle: false
      state_color: true
      entities:
      - switch.garagedoor
      - group.test_group

```
![test group](https://user-images.githubusercontent.com/29969334/81227686-accf6100-8fa1-11ea-98d5-d5102e994106.jpg)
![garagedoor within testgroup](https://user-images.githubusercontent.com/29969334/81227823-e1431d00-8fa1-11ea-9016-52fe218517d6.jpg)

## Javascript errors shown in your browser console/inspector

```txt

```

## Additional information
This seems to only affect switch template.
I tried moving this switch template to under light domain just for the sake of trying.
```
light.yaml                                     #### Changed this
- platform:template
  lights:                                         #### Changed this
    garagedoor:
      friendly_name: "Garage Door"
      value_template: '{{ is_state("cover.garagedoor","open") }}'
      turn_on:
        - service: cover.open_cover
          data: 
            entity_id: cover.garagedoor
      turn_off:
        - service: cover.close_cover
          data: 
            entity_id: cover.garagedoor


groups.yaml
test_group:
  name: test group
  entities:
  - switch.garagedoor
  - light.garagedoor                      #### Added this


ui-lovelace.yaml
views:
  - title: Test
    cards:
    - type: entities
      title: Test
      show_header_toggle: false
      state_color: true
      entities:
      - switch.garagedoor
      - light.garagedoor                   #### Added this
      - group.test_group
```
![light1](https://user-images.githubusercontent.com/29969334/81228434-ea80b980-8fa2-11ea-9031-8b5ab4ba1a19.jpg)
![light1 within testgroup](https://user-images.githubusercontent.com/29969334/81228581-27e54700-8fa3-11ea-8eca-505785585680.jpg)

**Comment by iantrich** [comment]
this should be fixed now that groups use state color

**Comment by xenz698** [comment]
It still hasn't fixed yet and I tested on HassOS 0.114.0
I did found a workaround though, which is moving the template from switch domain to a light domain.

This bug is about the switch.template within a group entities showing on lovelace.
